### PR TITLE
Docstring update: Clarify what "C-order" is

### PR DESCRIPTION
**Pull request by asmith26**
I didn't know what "C-order" was.After googling it I came across "column-major order" at [Wikipedia](https://en.wikipedia.org/wiki/Row-_and_column-major_order).

- [na] Tests added / passed
- [x] Passes `flake8 dask`

### CODE REVIEW

File: dask/array/reshape.py [hunk 1]
```diff
@@ -136,9 +136,11 @@ def reshape(x, shape):
     This is a parallelized version of the ``np.reshape`` function with the
     following limitations:
 
-    1.  It assumes that the array is stored in C-order
+    1.  It assumes that the array is stored in `column-major order`_
     2.  It only allows for reshapings that collapse or merge dimensions like
         ``(1, 2, 3, 4) -> (1, 6, 4)`` or ``(64,) -> (4, 4, 4)``
+        
+    .. _`column-major order`: https://en.wikipedia.org/wiki/Row-_and_column-major_order
```
[the remark this answers is not in the document]
Comment (asmith26) [reply dask/array/reshape.py]: Thanks for clarifying with me the issue.
I've recommitted and wrapped the link ([example](http://rst.ninjs.org/#ICAgIDEuICBJdCBhc3N1bWVzIHRoYXQgdGhlIGFycmF5IGlzIHN0b3JlZCBpbiBgY29sdW1uLW1ham9yIG9yZGVyYF8KICAgIDIuICBJdCBvbmx5IGFsbG93cyBmb3IgcmVzaGFwaW5ncyB0aGF0IGNvbGxhcHNlIG9yIG1lcmdlIGRpbWVuc2lvbnMgbGlrZQogICAgICAgIGBgKDEsIDIsIDMsIDQpIC0+ICgxLCA2LCA0KWBgIG9yIGBgKDY0LCkgLT4gKDQsIDQsIDQpYGAKICAgICAgICBgYCgxLCAyLCAzLCA0KSAtPiAoMSwgNiwgNClgYCBvciBgYCg2NCwpIC0+ICg0LCA0LCA0KWBgCiAgICAgICAgCiAgICAuLiBfYGNvbHVtbi1tYWpvciBvcmRlcmA6IGh0dHBzOi8vZW4ud2lraXBlZGlhLm9yZy93aWtpLwogICAgICAgICAgICAgICAgICAgICAgICAgICAgICBSb3ctX2FuZF9jb2x1bW4tbWFqb3Jfb3JkZXI=))

**Comment by TomAugspurger** [comment]
Still have a linting error: https://travis-ci.org/dask/dask/jobs/487619660#L2052

You can run flake8 locally FYI.

Do you know if sphinx will correctly render that link? Will it issue a warning?

It may be easier to shorten the text as @martindurant suggested.

**Comment by asmith26** [comment]
> Still have a linting error: https://travis-ci.org/dask/dask/jobs/487619660#L2052
> 
> You can run flake8 locally FYI.

Thanks for highlighting where the error is, and for the tip about flake 8 - should be fixed this time.
 
> Do you know if sphinx will correctly render that link? Will it issue a warning?

I built the doc locally with `make html`, and all looks good:

image


> It may be easier to shorten the text as @martindurant suggested.

I'm happy either way.

**Comment by mrocklin** [comment]
Is column-major the right term here?  By C ordering we mean the ordering
that the C language uses, wherein latter indices move fastest as we cycle
through linear address space.

Column-major vs row-major order seems to be more applicable in the 2d
matrix case?

**Comment by asmith26** [comment]
> Is column-major the right term here? By C ordering we mean the ordering that the C language uses, wherein latter indices move fastest as we cycle through linear address space. 

Thanks for clarification; in that case I actually think "row-major order" is more applicable here:
image

> Column-major vs row-major order seems to be more applicable in the 2d matrix case?

I can see where you're coming from; indeed, to quote Wikipedia:

 _While the terms allude to the rows and columns of a two-dimensional array, i.e. a matrix, **the orders can be generalized to arrays of any dimension** by noting that the terms row-major and column-major are equivalent to lexicographic and colexicographic orders, respectively._

Thus, it could be more useful to clarify the order/manner that dask reshapes is [the same as numpy's default setting](https://docs.scipy.org/doc/numpy-1.15.0/reference/generated/numpy.reshape.html) (I never realised numpy's reshape actually had a "order" argument).

**Comment by jrbourbeau** [comment]
> Thus, it could be more useful to clarify the order/manner that dask reshapes is [the same as numpy's default setting](https://docs.scipy.org/doc/numpy-1.15.0/reference/generated/numpy.reshape.html)

@asmith26 I agree that adding a note about Dask `reshape` assuming C-order, which is the NumPy `reshape` default, could help clarify things here. For what it's worth, the [NumPy `reshape` documentation](https://docs.scipy.org/doc/numpy/reference/generated/numpy.reshape.html#numpy.reshape), which the Dask `reshape` documentation links to, has a nice description of the different `order` options.

File: dask/array/reshape.py [hunk 2]
```diff
@@ -136,10 +136,13 @@ def reshape(x, shape):
     This is a parallelized version of the ``np.reshape`` function with the
     following limitations:
 
-    1.  It assumes that the array is stored in C-order
+    1.  It assumes that the array is stored in `column-major order`_
```
Comment (jakirkham) [inline dask/array/reshape.py]: Maybe "stored in C-order (sometimes referred to as `column-major order`_)"?

Comment (jrbourbeau) [reply dask/array/reshape.py]: I think we decided C-order is row-major order. I like this suggestion, +1 for adding "(sometimes referred to as `row-major order`_)"

**Comment by mrocklin** [comment]
I've force pushed column -major to row-major.  Merging.

**Pull request merged**